src/whispercpp/api_export.cc

```diff
@@ -148,10 +150,18 @@ PYBIND11_MODULE(api, m) {
       .def_property("logprob_threshold", &Params::get_logprob_thold,
                     &Params::set_logprob_thold)
       .def_property("no_speech_threshold", &Params::get_no_speech_thold,
-                    &Params::set_no_speech_thold);
-  // TODO: idk what to do with setting all the callbacks for Params. API are
-  // there, but need more time investingating conversion from Python callback to
-  // C++ callback
-  // Callback has to be pure here. Otherwise this won't work.
+                    &Params::set_no_speech_thold)
+      .def("on_new_segment",
+           [](Params &self, NewSegmentCallback & callback, py::object & user_data) {
+             using namespace std::placeholders;
+             self.set_new_segment_callback(
+               std::bind([](NewSegmentCallback & callback, py::object & user_data,
+                            Context & ctx, int n_new) mutable {
```

Comment:
very nit here to keep it consistent

```suggestion
               std::bind([](NewSegmentCallback & callback, Context & ctx, int n_new, py::object & user_data) mutable {
```